subsys/jwt/Kconfig

```diff
@@ -16,19 +16,45 @@ choice
 
 config JWT_SIGN_RSA
 	bool "Use RSA signature (RS-256)"
-	depends on CSPRNG_ENABLED
-	select MBEDTLS
-	select MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
-	select PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY if PSA_CRYPTO_CLIENT
-	select PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_BASIC if PSA_CRYPTO_CLIENT
-	select PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_IMPORT if PSA_CRYPTO_CLIENT
 
 config JWT_SIGN_ECDSA
 	bool "Use ECDSA signature (ES-256)"
+
+endchoice
+
+if JWT_SIGN_ECDSA
+
+config JWT_SIGN_ECDSA_PSA
```

Comment:
Just curious, does the PSA configuration basically assume/assert that there is a proper CSPRNG enabled? Otherwise, should we check that. Signing multiple messages with the same nonce in EDCSA is a devastating failure.

Reply:
I would say that the answer is yes, but it might depend on your configuration/platform capabilities.
I've recently done some change on how the Mbed TLS's PSA can handle random generation. You have basically [2 choices](https://github.com/zephyrproject-rtos/zephyr/blob/ccc4357f0a2c4e1b2cc39f48c227f3dc905ba3d0/modules/mbedtls/Kconfig.tls-generic#L464):

- _legacy_, where `psa_generate_random()` calls to the legacy Mbed TLS modules (i.e. entropy +  ctr/hmac drbg) to get random values; in this case the entropy module [calls back ](https://github.com/zephyrproject-rtos/zephyr/blob/ccc4357f0a2c4e1b2cc39f48c227f3dc905ba3d0/modules/mbedtls/zephyr_entropy.c#L34)to Zephyr to poll some random values. It first tries with CS sources, then it falls back to non-CS ones.

- _external_, where `psa_generate_random()` calls to CS-only sources. In this case there is the possibility to use non-CS sources (via `CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`), but there are warning during the build that clearly state that this is meant to be used only as test.